src/AMAMM.sol

```diff
@@ -183,15 +185,9 @@ contract AMAMM is IAmAmm {
     }
 
     function _updateLastUpdatedEpoch(PoolId id, uint40 _epoch) internal returns (uint40) {
-        if (_lastUpdatedEpoch[id] > _getEpoch(id, block.timestamp)) {
-            if (_epoch < _lastUpdatedEpoch[id]) {
-                return _lastUpdatedEpoch[id] = _epoch;
-            } else {
-                return _lastUpdatedEpoch[id];
-            }
-        } else {
-            return _lastUpdatedEpoch[id] = _epoch;
-        }
+        //We know that _epoch - currEpoch <= k
```

Comment:
```suggestion

```
There's nothing to do with the `currentEpoch`.

Reply:
@leckylao I was referring to this check on line 92
``` _epoch > _getEpoch(id, block.timestamp)```
We know that the epoch will be <= K + currentEpoch